java/google/registry/flows/FlowMetadata.java

```diff
@@ -0,0 +1,24 @@
+package google.registry.flows;
+
+import com.google.auto.value.AutoValue;
+import google.registry.model.ImmutableObject;
+
+/** Object to allow setting and retrieving metadata information in flows. */
+@AutoValue
+public abstract class FlowMetadata extends ImmutableObject {
+
+  public abstract boolean superuser();
```

Comment:
This should have a line of javadoc, even though it's fairly self-evident, just for the sake of being explicit.  Something like "True if this flow is being run with superuser privileges."